File: Data Structures & Algorithms/longest-common-prefix/submission-9.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        
        Array.Sort(strs);

        string first=strs[0];
        string last=strs[strs.Length-1];

        string res="";

        for (int idx=0; idx<first.Length; idx++){
            if (first[idx]!=last[idx]){
                break;
            }
            res+=first[idx];
        }
        return res;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-9.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        encoded=""
        for ch in strs:
            encoded+=str(len(ch))+"#"+ch;

        return encoded

    def decode(self, s: str) -> List[str]:

        j=0
        i=0
        res=[]
        while i<len(s):
            j=i

            while s[j]!="#":
                j+=1

            length=int(s[i:j])
            start=j+1

            res.append(s[start:start+length])
            i=start+length

        return res
